app/instrumentation-tests/src/org/commcare/androidTests/LogSubmissionTest.kt

```diff
@@ -42,4 +51,25 @@ class LogSubmissionTest: BaseTest() {
                 "Log submission was unsuccessful"
         )
     }
+
+    @Test
+    fun testReportProblem() {
+        InstrumentationUtility.selectOptionItem(withText("Advanced"))
+        onView(withText("Report Problem"))
+            .perform(click())
+        InstrumentationUtility.enterText(R.id.ReportText01,"This is a test for Report Problem")
+        InstrumentationUtility.hideKeyboard()
+        onView(withText("Submit Report")).isPresent()
+        InstrumentationUtility.rotateLeft()
+        InstrumentationUtility.matchTypedText(R.id.ReportText01,"This is a test for Report Problem")
+        InstrumentationUtility.hideKeyboard()
+        onView(withText("Submit Report")).isPresent()
+        InstrumentationUtility.rotatePortrait()
+        InstrumentationUtility.hideKeyboard()
+        onView(withText("Submit Report")).perform(click())
+        InstrumentationUtility.stubIntentWithAction(Intent.ACTION_SEND)
+        InstrumentationUtility.verifyIntentText("This is a test for Report Problem")
```

Comment:
this is not doing anything right now as you are not adding the intent callback generated by `verifyIntentText` to `IntentMonitorRegistry`. Also addition of callback needs to happen before we click on the submit button so that we can capture intent when it's fired on clicking submit. This is the code you need to have instead -  

````
        var intentcallback = verifyIntent(inputText)
        IntentMonitorRegistry.getInstance().addIntentCallback(intentcallback)
        onView(withText("Submit Report")).perform(click())
        IntentMonitorRegistry.getInstance().removeIntentCallback(intentcallback)
    }


    fun verifyIntent(inputText: String) = IntentCallback { intent ->
        var extraText = intent.extras!!.getString(Intent.EXTRA_TEXT)
        assertTrue(extraText!!.contains(inputText))
        assertTrue(intent.action.contentEquals(Intent.ACTION_SEND))
    }
````

Reply:
Renamed the earlier function that validated the callback to verifyIntentTextOnSubmitReport() and added all the steps in the same function. I tried to add the steps in the test file but was getting error of "inferred type" so added them in the utility file.

Reply:
Also removed InstrumentationUtility.stubIntentWithAction(Intent.ACTION_SEND)